backend/pikpak/pikpak.go

```diff
@@ -0,0 +1,1749 @@
+// entries can be returned in any order but should be for a
+// complete directory.
+//
+// dir should be "" to list the root, and should not have
+// trailing slashes.
+//
+// This should return ErrDirNotFound if the directory isn't
+// found.
+func (f *Fs) List(ctx context.Context, dir string) (entries fs.DirEntries, err error) {
+	// fs.Debugf(f, "List(%q)\n", dir)
+	dirID, err := f.dirCache.FindDir(ctx, dir, false)
+	if err != nil {
+		return nil, err
+	}
+	var iErr error
+	trashed := "false"
+	if f.opt.TrashedOnly {
+		// still need to list folders
+		trashed = ""
+	}
+	_, err = f.listAll(ctx, dirID, "", trashed, func(item *api.File) bool {
+		entry, err := f.itemToDirEntry(ctx, path.Join(dir, item.Name), item)
+		if err != nil {
+			iErr = err
+			return true
+		}
+		if entry != nil {
+			entries = append(entries, entry)
+		}
+		return false
+	})
+	if err != nil {
+		return nil, err
+	}
+	if iErr != nil {
+		return nil, iErr
+	}
+	return entries, nil
+}
+
+// CreateDir makes a directory with pathID as parent and name leaf
+func (f *Fs) CreateDir(ctx context.Context, pathID, leaf string) (newID string, err error) {
+	// fs.Debugf(f, "CreateDir(%q, %q)\n", pathID, leaf)
+	req := api.RequestNewFile{
+		Name:     f.opt.Enc.FromStandardName(leaf),
+		Kind:     api.KindOfFolder,
+		ParentID: parentIDForRequest(pathID),
+	}
+	info, err := f.requestNewFile(ctx, &req)
+	if err != nil {
+		return "", err
+	}
+	return info.File.ID, nil
+}
+
+// Mkdir creates the container if it doesn't exist
+func (f *Fs) Mkdir(ctx context.Context, dir string) error {
+	_, err := f.dirCache.FindDir(ctx, dir, true)
+	return err
+}
+
+// About gets quota information
+func (f *Fs) About(ctx context.Context) (usage *fs.Usage, err error) {
+	info, err := f.getAbout(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get drive#quota: %w", err)
+	}
+	q := info.Quota
+	usage = &fs.Usage{
+		Used: fs.NewUsageValue(q.Usage), // bytes in use
+		// Trashed: fs.NewUsageValue(q.UsageInTrash), // bytes in trash but this seems not working
+	}
+	if q.Limit > 0 {
+		usage.Total = fs.NewUsageValue(q.Limit)          // quota of bytes that can be used
+		usage.Free = fs.NewUsageValue(q.Limit - q.Usage) // bytes which can be uploaded before reaching the quota
+	}
+	return usage, nil
+}
+
+// PublicLink adds a "readable by anyone with link" permission on the given file or folder.
+func (f *Fs) PublicLink(ctx context.Context, remote string, expire fs.Duration, unlink bool) (string, error) {
+	// TODO: can be improved further; expiration days, passcode, and ...
+	id, err := f.dirCache.FindDir(ctx, remote, false)
+	if err == nil {
+		fs.Debugf(f, "attempting to share directory '%s'", remote)
+	} else {
+		fs.Debugf(f, "attempting to share single file '%s'", remote)
+		o, err := f.NewObject(ctx, remote)
+		if err != nil {
+			return "", err
+		}
+		id = o.(*Object).id
+	}
+	req := api.RequestShare{
+		FileIds:        []string{id},
+		ShareTo:        "publiclink",
+		ExpirationDays: -1,
```

Comment:
You can calculate the `ExpirationDays` from the `expire` duration passed in I think.